src-tauri/src/services/network_service.rs

```diff
@@ -1,17 +1,39 @@
-use std::{sync::Arc, time::Duration};
+use std::{fmt::Display, sync::Arc, time::Duration};
 
 use async_trait::async_trait;
 use tauri::async_runtime::Mutex;
 use tauri_plugin_http::reqwest;
 
 use crate::config;
 
-#[derive(Debug, PartialEq, Eq)]
+#[derive(Debug, PartialEq, Clone, Copy, Eq)]
+pub enum FrontEndEvent {
+    NetworkMode,
+}
+
+impl Display for FrontEndEvent {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            FrontEndEvent::NetworkMode => write!(f, "NetworkMode"),
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Clone, Copy, Eq)]
 pub enum NetworkMode {
     Local,
     Remote,
 }
 
+impl Display for NetworkMode {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            NetworkMode::Local => write!(f, "Local"),
+            NetworkMode::Remote => write!(f, "Remote"),
+        }
+    }
+}
```

Comment:
Why isn't this actually implemented?

Also we should probably get 2 separate events, like "NetworkActivated" and "NetworkDeactivated", or whatever better sounding name you may find?

Reply:
Having it in a enum that implements Display is enough to represent an event that will be sent to the frontend.